### PR TITLE
Feature/ae 67

### PR DESCRIPTION
**Pull request by hkorpi**

### CODE REVIEW

File: etp-backend/src/main/clj/solita/etp/db.clj [hunk 1]
```diff
@@ -11,10 +13,35 @@
...
+          (case (.getSQLState error)
+            "23505"
+            (ex-info
+              (.getMessage psqle)
+              {:type       :unique-violation
+               :constraint (keyword (str/replace (.getConstraint error) "_" "-"))}
+              psqle)
+            psqle))))))
+
+(def original-generate-query-fn jeesql.generate/generate-query-fn)
```
Comment (MattiNieminen) [inline etp-backend/src/main/clj/solita/etp/db.clj]: I would be good to require this namespace

File: etp-backend/src/main/clj/solita/etp/service/json.clj [hunk 2]
```diff
@@ -13,3 +13,7 @@
 (defn merge-data [db-row]
   (let [data (-> db-row :data .getValue read-value)]
     (merge data (dissoc db-row :data))))
+
+(defn data-db-row [object & keep]
```
Comment (MattiNieminen) [inline etp-backend/src/main/clj/solita/etp/service/json.clj:17]: :+1:

File: etp-backend/src/test/clj/solita/etp/service/yritys_test.clj [hunk 3]
```diff
@@ -1,14 +1,30 @@
...
+            [schema-generators.complete :as c]
             [solita.etp.test-system :as ts]
             [solita.etp.service.yritys :as service]
-            [solita.etp.schema.yritys :as schema]))
+            [solita.etp.schema.yritys :as schema]
+            [solita.etp.schema.common :as common-schema]))
 
 (t/use-fixtures :each ts/fixture)
 
+(def counter (atom 0))
```
Comment (MattiNieminen) [inline etp-backend/src/test/clj/solita/etp/service/yritys_test.clj]: Would be nice to get rid of atom on namespace level

**Review by MattiNieminen — APPROVED** [review]

File: etp-backend/src/test/clj/solita/etp/schema/common_test.clj [hunk 4]
```diff
@@ -12,3 +12,12 @@
...
   (t/is (nil? (schema/check common/Henkilotunnus "130200A892S"))))
+
+
+(t/deftest valid-ytunnus?-test
+  (t/is (nil? (schema/check common/Ytunnus "1234567-1")))
+  (t/is (some? (schema/check common/Ytunnus "1234567-2")))
+  (t/is (nil? (schema/check common/Ytunnus "1060155-5")))
+  (t/is (some? (schema/check common/Ytunnus "1060155-6")))
+  (t/is (some? (schema/check common/Ytunnus "1060155-7")))
+  (t/is (nil? (schema/check common/Ytunnus "0000001-9"))))
```
Comment (solita-tonipu) [inline etp-backend/src/test/clj/solita/etp/schema/common_test.clj]: Add a test case for an invalid y-tunnus with containing alphabets

**Review by solita-tonipu — APPROVED** [review]

**Pull request merged**